# Enerji Perakende Veri Analizi – Veri Keşfi ve Tanımlayıcı İstatistik

## Amaç
Bu çalışmanın amacı; Amasya iline bağlı Hamamözü, Gümüşhacıköy ve Göynücek ilçelerine ait
elektrik tüketim (tahakkuk) ve ödeme (tahsilat) verilerini analiz ederek:

- Verinin yapısını ve kalitesini anlamak
- Temel istatistiksel göstergeleri incelemek
- Olası veri problemlerini (eksik, negatif, aykırı değerler) tespit etmek
- İlerleyen analizler için güvenilir bir veri zemini oluşturmaktır

Bu notebook, yönetime sunulacak ileri analizler için hazırlık ve doğrulama niteliğindedir.


In [2]:
import pandas as pd
import numpy as np


Bu bölümde analiz boyunca kullanılacak temel Python kütüphaneleri yüklenmiştir.

- pandas: Tablo yapısındaki verileri okumak ve analiz etmek için
- numpy: Sayısal hesaplamalar için


In [8]:
file_path = "//content/elektrik_veri.xlsx"
xls = pd.ExcelFile(file_path)
xls.sheet_names

['Tahsilat', 'Tahsilat 1', 'Tahakkuk', 'Tahakkuk 1', 'Tahakkuk 2']

In [9]:
df_tahsilat = pd.read_excel(xls, sheet_name="Tahsilat")
df_tahsilat_1 = pd.read_excel(xls, sheet_name="Tahsilat 1")
df_tahakkuk = pd.read_excel(xls, sheet_name="Tahakkuk")
df_tahakkuk_1 = pd.read_excel(xls, sheet_name="Tahakkuk 1")
df_tahakkuk_2 = pd.read_excel(xls, sheet_name="Tahakkuk 2")


In [10]:
print("Tahsilat:", df_tahsilat.shape)
print("Tahsilat 1:", df_tahsilat_1.shape)
print("Tahakkuk:", df_tahakkuk.shape)
print("Tahakkuk 1:", df_tahakkuk_1.shape)
print("Tahakkuk 2:", df_tahakkuk_2.shape)


Tahsilat: (636993, 9)
Tahsilat 1: (917632, 22)
Tahakkuk: (124818, 10)
Tahakkuk 1: (765657, 10)
Tahakkuk 2: (295223, 10)


## Veri Yapısının İncelenmesi

Bu bölümde her bir veri setinin:

- Sütun sayısı ve veri tipleri
- Eksik değer durumu
- Sayısal değişkenlerin temel istatistikleri
- İlk birkaç gözlem üzerinden genel yapısı

incelenmiştir.

Amaç, analiz öncesinde verinin güvenilirliğini ve analize uygunluğunu
değerlendirmektir.


In [11]:
df_tahsilat.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 636993 entries, 0 to 636992
Data columns (total 9 columns):
 #   Column                 Non-Null Count   Dtype         
---  ------                 --------------   -----         
 0   Şube                   636993 non-null  object        
 1   Kasa                   636993 non-null  object        
 2   İlçe                   636993 non-null  object        
 3   Söz.hsp.(bağımsız)     636993 non-null  int64         
 4   Tahsilat Tarihi        636993 non-null  datetime64[ns]
 5   Nakit Tahsilat         523 non-null     float64       
 6   Mahsuben Tahsilat      7542 non-null    float64       
 7   Kredi Kartı Tahsilatı  0 non-null       float64       
 8   Banka Tahsilatı        628933 non-null  float64       
dtypes: datetime64[ns](1), float64(4), int64(1), object(3)
memory usage: 43.7+ MB


In [12]:
df_tahakkuk.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 124818 entries, 0 to 124817
Data columns (total 10 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   il                 124818 non-null  object 
 1   ilce               124818 non-null  object 
 2   sozlesme_hesap_no  124818 non-null  int64  
 3   mali_yil_donem     124818 non-null  object 
 4   fatura_tarihi      124818 non-null  object 
 5   kayit_tarihi       124818 non-null  object 
 6   vade_tarihi        124818 non-null  object 
 7   hesap_sinifi       124818 non-null  object 
 8   Hesap Sınıfı       124818 non-null  object 
 9   kwh                124818 non-null  float64
dtypes: float64(1), int64(1), object(8)
memory usage: 9.5+ MB


In [13]:
df_tahakkuk_1.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 765657 entries, 0 to 765656
Data columns (total 10 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   il                 765657 non-null  object 
 1   ilce               765657 non-null  object 
 2   sozlesme_hesap_no  765657 non-null  int64  
 3   mali_yil_donem     765657 non-null  object 
 4   fatura_tarihi      765657 non-null  object 
 5   kayit_tarihi       765657 non-null  object 
 6   vade_tarihi        765657 non-null  object 
 7   hesap_sinifi       765657 non-null  object 
 8   Hesap Sınıfı       765657 non-null  object 
 9   kwh                765657 non-null  float64
dtypes: float64(1), int64(1), object(8)
memory usage: 58.4+ MB


In [14]:
df_tahakkuk_2.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 295223 entries, 0 to 295222
Data columns (total 10 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   il                 295223 non-null  object 
 1   ilce               295223 non-null  object 
 2   sozlesme_hesap_no  295223 non-null  int64  
 3   mali_yil_donem     295223 non-null  object 
 4   fatura_tarihi      295223 non-null  object 
 5   kayit_tarihi       295223 non-null  object 
 6   vade_tarihi        295223 non-null  object 
 7   hesap_sinifi       295223 non-null  object 
 8   Hesap Sınıfı       295223 non-null  object 
 9   kwh                295223 non-null  float64
dtypes: float64(1), int64(1), object(8)
memory usage: 22.5+ MB


### Veri Yapısı Değerlendirmesi (.info)

**Tahsilat Verisi**
- Toplam 636.993 satır ve 9 sütundan oluşmaktadır.
- Ödeme tarihi alanı `Tahsilat Tarihi` tarih formatındadır ve zaman bazlı analizlere uygundur.
- Tüm temel tanımlayıcı alanlarda (Şube, Kasa, İlçe, Sözleşme Hesap No) eksik veri bulunmamaktadır.
- Ödeme türlerine bakıldığında:
  - Banka tahsilatı baskın ödeme kanalıdır.
  - Nakit ve mahsup ödemeleri oldukça sınırlıdır.
  - Kredi kartı tahsilatı veride yer almamaktadır; bu durum ya kanalın aktif olmadığını
    ya da veriye dahil edilmediğini düşündürmektedir.
- Eksik değerler ödeme türü bazında **yapısal** olup, veri kalitesi problemi olarak
  değil ödeme kanalına özgü doğal bir durum olarak değerlendirilmiştir.

**Tahakkuk (Tüketim) Verileri**
- Üç ilçe için ayrı tablolarda tutulan tüketim verilerinde:
  - Hamamözü: 124.818 kayıt
  - Gümüşhacıköy: 765.657 kayıt
  - Göynücek: 295.223 kayıt bulunmaktadır.
- Tüm tahakkuk tablolarında sütun yapısı birebir aynıdır; bu durum verilerin
  birleştirilerek analiz edilmesine olanak sağlamaktadır.
- `kwh` sütunu sayısal formatta olup, tüketim analizleri için uygundur.
- Tüm tarih alanları (mali dönem, fatura tarihi, vade tarihi vb.) object formatındadır;
  zaman serisi ve mevsimsellik analizleri öncesinde tarih tipine dönüştürülmeleri gerekmektedir.
- Tahakkuk verilerinde eksik gözlem bulunmaması, tüketim analizleri açısından
  güçlü ve tutarlı bir veri yapısına işaret etmektedir.

**Genel Değerlendirme**
- Veri setleri, proje dokümanında belirtilen kayıt sayılarıyla uyumludur.
- Analiz için kritik alanlarda eksik veri bulunmaması, çalışmanın güvenilirliğini artırmaktadır.
- Veri yapısı, ilerleyen aşamalarda karşılaştırmalı ilçe analizi, müşteri segmentasyonu
  ve ödeme davranışı incelemeleri için uygundur.


In [15]:
df_tahsilat.describe()


,Söz.hsp.(bağımsız),Tahsilat Tarihi,Nakit Tahsilat,Mahsuben Tahsilat,Kredi Kartı Tahsilatı,Banka Tahsilatı
count,6.369930e+05,636993,523.000000,7542.000000,0.0,628933.000000
mean,5.019884e+09,2024-03-05 09:26:07.644856320,694.966635,6180.182282,NaN,372.629109
min,1.758320e+05,2023-01-01 00:00:00,7.450000,-34508.950000,NaN,0.010000
25%,2.532887e+09,2023-07-28 00:00:00,425.330000,44.477500,NaN,120.000000
50%,5.008502e+09,2024-02-26 00:00:00,524.670000,290.410000,NaN,208.000000
75%,7.525722e+09,2024-09-30 00:00:00,688.830000,2729.110000,NaN,322.000000
max,9.999600e+09,2025-05-31 00:00:00,11373.740000,399526.780000,NaN,606473.800000
std,2.884435e+09,NaN,758.319428,23828.022593,NaN,3265.430202


In [16]:
df_tahakkuk.describe()
df_tahakkuk_1.describe()
df_tahakkuk_2.describe()


,sozlesme_hesap_no,kwh
count,2.952230e+05,295223.000000
mean,4.950566e+09,89.669891
std,2.887724e+09,742.276369
min,1.118550e+06,-4208.640000
25%,2.432854e+09,17.860000
50%,4.928206e+09,45.090000
75%,7.475616e+09,77.140000
max,9.997185e+09,105687.690000


### Tanımlayıcı İstatistikler (.describe) – Değerlendirme

`.describe()` çıktıları incelendiğinde aşağıdaki noktalar öne çıkmaktadır:

- `sozlesme_hesap_no` ve `Söz.hsp.(bağımsız)` alanları müşteri kimlik numarası
  niteliğinde olup, sayısal görünmelerine rağmen ölçüm değişkeni değildir.
  Bu nedenle bu sütunlara ait ortalama, standart sapma, minimum ve maksimum
  değerler analitik olarak anlam taşımamaktadır.
  Bu alanlar yalnızca benzersiz müşteri sayısı gibi sayım temelli analizlerde
  kullanılacaktır.

- Tahakkuk (tüketim) verilerinde `kwh` değişkeni için:
  - Ortalama değerlerin medyandan belirgin şekilde yüksek olması,
    tüketim dağılımının sağa çarpık olduğunu göstermektedir.
  - Standart sapmanın ortalamaya kıyasla oldukça yüksek olması,
    yüksek tüketimli az sayıda müşteri veya aykırı değerlerin varlığına işaret etmektedir.
  - Negatif minimum kWh değerleri gözlemlenmiştir; bu durum sayaç düzeltmeleri,
    iptal edilen faturalar veya geçmiş dönem düzeltmeleri kaynaklı olabilir.

- Tahsilat verilerinde:
  - Banka tahsilatı tutarlarının diğer ödeme kanallarına kıyasla baskın olduğu,
  - Mahsuben tahsilat ve nakit tahsilat tutarlarında yüksek varyans bulunduğu
    gözlemlenmiştir.

Bu bulgular, ilerleyen analizlerde aykırı değerlerin ayrıca ele alınması ve
kimlik niteliğindeki değişkenlerin istatistiksel yorumlara dahil edilmemesi
gerektiğini göstermektedir.


## Veri Yapısının Anlaşılması (.head)

Bu bölümde her bir veri setinin ilk birkaç satırı incelenerek,
her satırın neyi temsil ettiği ve veri alanlarının birbirleriyle
olan ilişkisi anlaşılmaya çalışılmıştır.

Bu inceleme, analiz sırasında yapılacak yorumların doğru temellere
dayanmasını sağlamak amacıyla gerçekleştirilmiştir.


In [17]:
df_tahsilat.head()


,Şube,Kasa,İlçe,Söz.hsp.(bağımsız),Tahsilat Tarihi,Nakit Tahsilat,Mahsuben Tahsilat,Kredi Kartı Tahsilatı,Banka Tahsilatı
0,Tayin edilmedi,Tayin edilmedi,TAŞOVA,4989745446,2023-11-06,NaN,8648.95,NaN,NaN
1,Tayin edilmedi,Tayin edilmedi,TAŞOVA,4989745446,2024-06-26,NaN,762.40,NaN,NaN
2,Tayin edilmedi,Tayin edilmedi,TAŞOVA,4989745446,2024-07-10,NaN,311.60,NaN,NaN
3,PTT,PTT/PV,TAŞOVA,4254955886,2023-01-19,NaN,NaN,NaN,130.5
4,PTT,PTT/PV,TAŞOVA,4254955886,2023-02-17,NaN,NaN,NaN,117.0


In [18]:
df_tahakkuk.head()


,il,ilce,sozlesme_hesap_no,mali_yil_donem,fatura_tarihi,kayit_tarihi,vade_tarihi,hesap_sinifi,Hesap Sınıfı,kwh
0,AMASYA,HAMAMÖZÜ,917576806,2023-01-01,2023-01-12,2023-03-06,2023-01-23,M001,Mesken,1.79
1,AMASYA,HAMAMÖZÜ,917576806,2023-01-01,2023-02-09,2023-05-11,2023-02-20,M001,Mesken,2.60
2,AMASYA,HAMAMÖZÜ,917576806,2023-02-01,2023-02-09,2023-05-11,2023-02-20,M001,Mesken,1.23
3,AMASYA,HAMAMÖZÜ,917576806,2023-02-01,2023-03-10,2023-05-11,2023-03-20,M001,Mesken,2.56
4,AMASYA,HAMAMÖZÜ,917576806,2023-03-01,2023-03-10,2023-05-11,2023-03-20,M001,Mesken,1.35


In [19]:
df_tahakkuk_1.head()


,il,ilce,sozlesme_hesap_no,mali_yil_donem,fatura_tarihi,kayit_tarihi,vade_tarihi,hesap_sinifi,Hesap Sınıfı,kwh
0,AMASYA,GÜMÜŞHACIKÖY,7444449517,2023-01-01,2023-01-11,2023-03-06,2023-01-23,M001,Mesken,21.85
1,AMASYA,GÜMÜŞHACIKÖY,7444449517,2023-01-01,2023-02-10,2023-05-11,2023-02-20,M001,Mesken,44.50
2,AMASYA,GÜMÜŞHACIKÖY,7444449517,2023-02-01,2023-02-10,2023-05-11,2023-02-20,M001,Mesken,22.25
3,AMASYA,GÜMÜŞHACIKÖY,7444449517,2023-02-01,2023-03-10,2023-05-11,2023-03-20,M001,Mesken,45.71
4,AMASYA,GÜMÜŞHACIKÖY,7444449517,2023-03-01,2023-03-10,2023-05-11,2023-03-20,M001,Mesken,25.40


In [20]:
df_tahakkuk_2.head()


,il,ilce,sozlesme_hesap_no,mali_yil_donem,fatura_tarihi,kayit_tarihi,vade_tarihi,hesap_sinifi,Hesap Sınıfı,kwh
0,AMASYA,GÖYNÜCEK,9374624783,2023-01-01,2023-01-14,2023-03-06,2023-01-24,M001,Mesken,0.10
1,AMASYA,GÖYNÜCEK,9374624783,2023-01-01,2025-03-12,2025-05-09,2025-03-24,M001,Mesken,0.12
2,AMASYA,GÖYNÜCEK,9374624783,2023-02-01,2025-03-12,2025-05-09,2025-03-24,M001,Mesken,0.20
3,AMASYA,GÖYNÜCEK,9374624783,2023-03-01,2025-03-12,2025-05-09,2025-03-24,M001,Mesken,0.00
4,AMASYA,GÖYNÜCEK,9374624783,2023-04-01,2025-03-12,2025-05-09,2025-03-24,M001,Mesken,0.08


###  Veri Yapısının Anlaşılması (.head)

**Tahsilat Verisi**
- Tahsilat tablosunda her satır, bir müşterinin belirli bir tarihte gerçekleştirdiği
  tek bir ödeme işlemini temsil etmektedir.
- Aynı müşteriye ait (Söz.hsp.(bağımsız)) birden fazla ödeme kaydı bulunabilmektedir;
  bu durum müşterilerin faturalarını farklı tarihlerde ve parçalı şekilde
  ödeyebildiğini göstermektedir.
- Ödeme tutarları, ödeme kanalına göre ayrı sütunlarda tutulmaktadır.
  İncelenen örnek kayıtlarda banka ve mahsuben tahsilatın öne çıktığı,
  kredi kartı tahsilatının ise fiilen kullanılmadığı görülmektedir.
- Şube ve kasa bilgilerinin bazı kayıtlarda “Tayin edilmedi” olarak yer alması,
  ödeme işlemlerinin belirli bir fiziksel şube yerine merkezi veya alternatif
  kanallar üzerinden gerçekleştiğine işaret etmektedir.

**Tahakkuk Verisi – Hamamözü**
- Tahakkuk tablolarında her satır, ilgili müşterinin belirli bir aya ait
  elektrik tüketim (kWh) kaydını göstermektedir.
- Aynı müşteri için aynı ay içerisinde birden fazla kayıt bulunabilmesi,
  fatura düzenleme veya dönemsel düzeltmelerin yapılabildiğini düşündürmektedir.
- Tüketim değerlerinin düşük seviyelerde yoğunlaştığı,
  ilçenin genel tüketim profilinin görece düşük olduğunu göstermektedir.

**Tahakkuk Verisi – Gümüşhacıköy**
- Veri yapısı Hamamözü ilçesi ile birebir aynıdır.
- Aynı müşteri için ardışık aylarda düzenli tüketim kayıtları bulunmaktadır.
- İlk gözlemler, Gümüşhacıköy ilçesinde Hamamözü’ne kıyasla
  daha yüksek tüketim seviyelerinin görülebildiğine işaret etmektedir.

**Tahakkuk Verisi – Göynücek**
- Veri yapısı diğer ilçelerle tutarlıdır ve karşılaştırmalı analiz için uygundur.
- Bazı müşterilerde çok düşük veya sıfıra yakın kWh değerleri gözlemlenmektedir;
  bu durum mevsimsel kullanım, geçici abonelik, kullanılmayan tesisler
  veya ölçüm/düzeltme kayıtları ile ilişkili olabilir.
- Bu tür düşük tüketim değerlerinin ilerleyen analizlerde ayrıca
  değerlendirilmesi gerekmektedir.

**Genel Değerlendirme**
- Tüm veri setlerinde satır bazlı anlam net olup,
  tahsilat ve tüketim verileri müşteri bazlı analizler için
  güçlü bir temel sunmaktadır.
- İlçe bazlı tahakkuk verilerinin yapısal olarak tutarlı olması,
  bölgesel karşılaştırmaların sağlıklı şekilde yapılabilmesini sağlamaktadır.


## İlçe Bazlı Müşteri Sayılarının Karşılaştırılması

Bu bölümde, her ilçeye ait benzersiz müşteri sayıları
sozlesme_hesap_no değişkeni üzerinden hesaplanmıştır.

Amaç, ilçelerin müşteri büyüklüklerini karşılaştırarak
ilerleyen tüketim ve tahsilat analizleri için
sağlam bir referans noktası oluşturmaktır.


In [23]:
hamamozu_musteri = df_tahakkuk["sozlesme_hesap_no"].nunique()
gumushacikoy_musteri = df_tahakkuk_1["sozlesme_hesap_no"].nunique()
goynucek_musteri = df_tahakkuk_2["sozlesme_hesap_no"].nunique()
hamamozu_musteri, gumushacikoy_musteri, goynucek_musteri
musteri_sayilari = pd.DataFrame({
    "İlçe": ["Hamamözü", "Gümüşhacıköy", "Göynücek"],
    "Benzersiz Müşteri Sayısı":[hamamozu_musteri,gumushacikoy_musteri,goynucek_musteri]})
musteri_sayilari



,İlçe,Benzersiz Müşteri Sayısı
0,Hamamözü,2981
1,Gümüşhacıköy,18190
2,Göynücek,7128


### Bulgular ve Yorum

- Gümüşhacıköy ilçesi 18.190 benzersiz müşteri ile
  analiz kapsamındaki ilçeler arasında açık ara
  en büyük müşteri tabanına sahiptir.
- Göynücek ilçesi 7.128 müşteri ile orta ölçekli bir
  müşteri profili sunmaktadır.
- Hamamözü ilçesi ise 2.981 müşteri ile
  diğer ilçelere kıyasla daha sınırlı bir
  müşteri tabanına sahiptir.
- İlçeler arasındaki bu belirgin müşteri sayısı farkı,
  toplam tüketim ve tahsilat tutarlarının
  doğrudan karşılaştırılmasının yanıltıcı olabileceğini
  göstermektedir.
- Bu nedenle ilerleyen analizlerde,
  müşteri başına düşen tüketim ve ödeme
  göstergeleri üzerinden değerlendirme yapılması
  daha sağlıklı olacaktır.


## Tahakkuk Veri Setlerinin Birleştirilmesi

Bu bölümde Hamamözü, Gümüşhacıköy ve Göynücek ilçelerine ait
tahakkuk veri setleri tek bir veri çerçevesi altında birleştirilmiştir.

Amaç, tüm ilçeleri kapsayan bütüncül bir tahakkuk veri seti oluşturarak
ilerleyen analizlerin tek bir tablo üzerinden gerçekleştirilmesini sağlamak ve toplam kayıt sayısını doğrulamaktır.


In [26]:
df_tahakkuk_all = pd.concat(
    [df_tahakkuk, df_tahakkuk_1, df_tahakkuk_2],
    axis=0,ignore_index=True)
df_tahakkuk_all.shape



(1185698, 10)

### Birleştirme Sonrası Kayıt Sayısı Kontrolü

Tahakkuk veri setlerinin birleştirilmesi sonucunda elde edilen
veri çerçevesi toplam 1.185.698 satır ve 10 sütundan oluşmaktadır.

Bu kayıt sayısı, her üç ilçeye ait tahakkuk veri setlerinin
satır sayılarının toplamına birebir eşittir.
Bu durum, birleştirme işlemi sırasında herhangi bir veri kaybı
ya da mükerrer ekleme olmadığını göstermektedir.

Sonuç olarak, oluşturulan birleşik tahakkuk veri seti
ilerleyen analizler için güvenle kullanılabilir durumdadır.


## kWh Değişkeninde Veri Kalitesi Kontrolü

Bu bölümde birleşik tahakkuk veri setinde yer alan
kwh değişkeni özelinde veri kalitesi kontrolü yapılmıştır.

Amaç, analiz sonuçlarını yanıltabilecek eksik değerleri,
negatif tüketimleri ve aşırı uç gözlemleri tespit ederek
ilerleyen adımlarda nasıl ele alınmaları gerektiğine
karar vermektir.


In [27]:
df_tahakkuk_all["kwh"].isna().sum()


np.int64(0)

In [28]:
(df_tahakkuk_all["kwh"] < 0).sum()


np.int64(151)

In [29]:
df_tahakkuk_all["kwh"].describe()


,kwh
count,1.185698e+06
mean,9.264207e+01
std,9.504052e+02
min,-2.537064e+04
25%,1.801000e+01
50%,4.662000e+01
75%,8.000000e+01
max,1.535757e+05


In [30]:
Q1 = df_tahakkuk_all["kwh"].quantile(0.25)
Q3 = df_tahakkuk_all["kwh"].quantile(0.75)
IQR = Q3 - Q1
alt_sinir = Q1 - 1.5 * IQR
ust_sinir = Q3 + 1.5 * IQR
outlier_sayisi = (
    (df_tahakkuk_all["kwh"] < alt_sinir) |
    (df_tahakkuk_all["kwh"] > ust_sinir)
).sum()
alt_sinir, ust_sinir, outlier_sayisi


(np.float64(-74.97499999999998),
 np.float64(172.98499999999999),
 np.int64(48554))

### kWh Değişkenine Ait Bulgular

Birleşik tahakkuk veri setinde yer alan `kwh` değişkeni üzerinde
yapılan inceleme sonucunda aşağıdaki bulgular elde edilmiştir:

- `kwh` sütununda eksik (NaN) değer bulunmamaktadır.
  Bu durum, elektrik tüketimi bilgisinin tüm kayıtlarda
  eksiksiz şekilde yer aldığını göstermektedir.
- Toplam 151 adet negatif kWh değeri tespit edilmiştir.
- Tanımlayıcı istatistiklere göre kWh değerleri geniş bir dağılım
  göstermektedir.
  Minimum değer -25.370,64 kWh iken,
  maksimum değer 153.575,70 kWh seviyesindedir.
- Çeyrekler arası aralık (IQR) yöntemi kullanılarak yapılan
  aşırı uç değer analizinde,
  alt sınır -74,97 kWh, üst sınır ise 172,98 kWh olarak hesaplanmıştır.
- Bu sınırların dışında kalan toplam 48.554 adet kayıt
  aşırı uç değer olarak tespit edilmiştir.

Bu bulgular, kWh değişkeninin analiz öncesinde
negatif ve aşırı uç değerler açısından ele alınması gerektiğini,
aksi takdirde ortalama ve toplam tüketim gibi metriklerin
yanıltıcı sonuçlar üretebileceğini göstermektedir.


## Hesap Sınıfına Göre Elektrik Tüketimlerinin İncelenmesi

Bu bölümde birleşik tahakkuk veri seti kullanılarak
hesap sınıflarına göre elektrik tüketim istatistikleri hesaplanmıştır.

Amaç, farklı abone türlerinin (hesap sınıflarının)
ortalama tüketim düzeylerini, tipik tüketim davranışlarını
(medyan) ve tüketim değişkenliğini
(standart sapma) karşılaştırmalı olarak incelemektir.


In [31]:
hesap_sinifi_istatistikleri = (
 df_tahakkuk_all
  .groupby("Hesap Sınıfı")["kwh"]
   .agg(
     ortalama_kwh="mean",
     medyan_kwh="median",
     std_kwh="std").reset_index())
hesap_sinifi_istatistikleri

,Hesap Sınıfı,ortalama_kwh,medyan_kwh,std_kwh
0,1 SAYILI CETVELDE YER ALAN KAMU İDARESİ,688.441598,23.860,3911.912974
1,Aritma Tesisleri,16594.174857,16186.910,11656.705924
2,Balıkçılık ve Su Ürünleri Yetiştiriciliğ,85.358306,79.375,42.120066
3,Belediye,600.581235,78.270,3350.978976
4,Belediye Park Bahçe Aydınlatma,31.951003,18.390,38.472449
5,"Bina Ort Kul (Asn,Hidr,Kapıcı Dai vb.)",29.965663,15.760,41.331395
6,Büyükbaş-Küçükbaş Hayvancılık,116.848021,24.110,408.549664
7,Cemevleri,10.713755,5.625,16.453958
8,Diyanet Kuran Kursu,167.135000,22.115,526.354274
9,Doğal Afet Barınma,73.615948,73.865,39.066907


### Hesap Sınıfına Göre Elektrik Tüketimi Bulguları

Hesap sınıfına göre hesaplanan ortalama, medyan ve standart sapma
değerleri incelendiğinde, elektrik tüketim davranışlarının
abonelik türlerine göre önemli ölçüde farklılaştığı görülmektedir.

- En yüksek ortalama ve medyan tüketim değerleri,
  **Karayolları Genel Müdürlüğü Aydınlatma**,
  **Arıtma Tesisleri** ve **Lisansız Üreticiler** hesap sınıflarında
  gözlemlenmiştir. Bu gruplar, yüksek ve sürekli enerji ihtiyacı olan
  altyapı ve üretim faaliyetlerini yansıtmaktadır.
- **Sanayi** ve **Tarımsal Faaliyetler** gibi üretim odaklı hesap
  sınıflarında ortalama tüketim değerleri yüksek olmakla birlikte,
  medyan değerlerin ortalamaya göre daha düşük olması,
  tüketimin sınırlı sayıdaki yüksek hacimli aboneler tarafından
  yukarı çekildiğini göstermektedir.
- Mesken hesap sınıfında ortalama (55,59 kWh) ve medyan
  (47,36 kWh) değerlerinin birbirine yakın olması,
  konut abonelerinde daha homojen bir tüketim yapısına işaret etmektedir.
- Kamu, belediye ve resmi kurumlara ait hesap sınıflarında
  standart sapma değerlerinin yüksek olması,
  bu gruplar içinde tüketim düzeylerinin oldukça değişken
  olduğunu göstermektedir.
- İbadethane, bina ortak kullanım ve benzeri düşük tüketimli
  hesap sınıflarında ortalama ve medyan değerlerin düşük,
  standart sapmanın sınırlı olduğu görülmektedir.

Genel olarak, standart sapma değerlerinin birçok hesap sınıfında
ortalama tüketim değerlerine kıyasla yüksek olması,
kWh değişkeninde uç değer etkisinin devam ettiğini göstermektedir.

